playbooks/openshift-node/private/join.yml

```diff
@@ -16,10 +17,35 @@
     - hostvars[item].openshift.common is defined
     - (hostvars[item].openshift_is_bootstrapped | bool) or (hostvars[item].openshift_node_bootstrap | default(True) | bool)
     changed_when: False
+  - name: Create a tempdir for node configs
+    command: mktemp -d /tmp/openshift-ansible-XXXXXXX
+    register: mktemp
+    changed_when: False
+
+- name: Copy bootstrap and CNI config
+  hosts: oo_first_master
+  tasks:
+  - name: Copy master bootstrap config locally
+    fetch:
+      src: "{{ openshift.common.config_base }}/master/bootstrap.kubeconfig"
+      dest: "{{ hostvars['localhost'].mktemp.stdout }}/bootstrap.kubeconfig"
+      flat: yes
+
+  - name: Find CNI config files
+    find:
+      path: /etc/cni/net.d/
+    register: cni_configs
+
+  - name: Copy master CNI settings
+    fetch:
+      src: "{{ item.path }}"
+      dest: "{{ hostvars['localhost'].mktemp.stdout }}/{{ item.path | basename }}"
+      flat: yes
+    with_items: "{{ cni_configs.files }}"
 
 - name: Gather facts from nodes to bootstrap
   hosts: oo_nodes_to_bootstrap
-  gather_facts: no
+  serial: 1
```

Comment:
Why would we do this serially?

Reply:
This could be a remainder from my experiments with manual CSR approver. Since now these are approved by `bootstrap-autoapprover` there is no need to start services sequencially